app/src/main/java/com/wut/indoornavigation/data/mesh/Mesh.java

```diff
@@ -0,0 +1,398 @@
+                return;
+            }
+
+            List<Stairs> endFloorStairs = kFloor.getStairs();
+
+            if (stairs.getEndfloor() == floorNumber) {
+                return;
+            }
+
+            List<Vertex> endFloorStairsGraphVertices = stairsVerticesDict.get(k);
+            int endVertexIndex = findMatchingStairsOnFloor(stairs, endFloorStairs);
+
+            Vertex startVertex = stairsVerticesDict.get(floorNumber).get(i);
+            Vertex endVertex = endFloorStairsGraphVertices.get(endVertexIndex);
+
+            listStairOnFloor(graph, startVertex, endVertex);
+        }
+    }
+
+    private void listStairOnFloor(Graph graph, Vertex startVertex, Vertex endVertex) {
+        if (startVertex.getId() != endVertex.getId()) {
+            if (!graph.containsEdge(startVertex.getId(), endVertex.getId())) {
+                graph.addEdge(new Edge(startVertex, endVertex, EDGE_STAIRS_WEIGHT));
+            }
+        }
+    }
+
+    private int findMatchingStairsOnFloor(Stairs stairs, List<Stairs> endFloorStairs) {
+        int endVertexIndex = -1;
+        for (int j = 0; j < endFloorStairs.size(); j++) {
+            if (endFloorStairs.get(j).getId() == stairs.getId()) {
+                endVertexIndex = j;
+                break;
+            }
+        }
+
+        if (endVertexIndex == -1) {
+            throw new IllegalStateException("This algorithm is bugged as f*ck.");
+        }
+        return endVertexIndex;
+    }
+
+    @Nullable
+    private Vertex processCell(int x, int y, FloorObject[][] enumMap, int floorNumber, boolean[][] visited, Graph graph) {
+        if (visited[x][y]) {
+            return graph.getVertexByCoordinates((float) x / 2, (float) y / 2, floorNumber);
+        }
+
+        visited[x][y] = true;
+        if (enumMap[x][y] == FloorObject.SPACE || enumMap[x][y] == FloorObject.DOOR || enumMap[x][y] == FloorObject.ROOM || enumMap[x][y] == FloorObject.STAIRS || enumMap[x][y] == FloorObject.ELEVATOR) {
+            Point coordinates = new Point((float) x / 2, (float) y / 2, floorNumber);
+            Vertex vertex = new Vertex(idSeed--, coordinates);
+            graph.addVertex(vertex);
+
+            List<Vertex> vertices = findVerticesSetByFloorObjectSign(enumMap[x][y], floorNumber);
+
+            if (vertices != null) {
+                vertices.add(vertex);
+            }
+
+            return vertex;
+        }
+
+        if (enumMap[x][y] == FloorObject.BEACON) {
+            Point coordinates = new Point((float) x / 2, (float) y / 2, floorNumber);
+            List<Point> floorBeacons = beaconsDict.get(floorNumber);
+            if (floorBeacons == null) {
+                floorBeacons = new ArrayList<>();
+                beaconsDict.put(floorNumber, floorBeacons);
+            }
+            floorBeacons.add(coordinates);
+        }
+
+        return null;
+    }
+
+    @Nullable
+    private List<Vertex> findVerticesSetByFloorObjectSign(FloorObject floorObject, int floorNumber) {
```

Comment:
probowalem to zrobic ale jakos nie widze komcepcji; probowalem tez zrobic jakby strategie processingu dla kazdego znaku osobno ale wyszlo mi ze to bez sensu bo w zasadzie to bardzo podobne jest tylko jedynie wierzcholek dodawany jest do innego zbioru. pogadamy o tym najwyzej albo ty to zrobisz :D bo ja widze strategie tylko w tych miejscach gdzie wykonuję (biznesowo) tą samą operację tylko w inny sposób, a tu tego nie widzę